### PR TITLE
[BEAM-1498] Use Flink-native side outputs

### PR DESCRIPTION
**Pull request by JingsongLi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by JingsongLi** [comment]
R: @aljoscha

**Comment by JingsongLi** [comment]
Run Flink ValidatesRunner

**Comment by aljoscha** [comment]
retest this please

**Comment by aljoscha** [comment]
Very good set of changes! 👍 

Could you also remove the `FnOutputT` parameter from `DoFnOperator` (in a separate commit). I quickly tried that locally and it works but I didn't want to steal from your PR. 😅

**Comment by JingsongLi** [comment]
Yes, now FnOutputT and OutputT are always the same.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java [hunk 1]
```diff
@@ -94,21 +94,21 @@
  * Flink operator for executing {@link DoFn DoFns}.
  *
  * @param <InputT> the input type of the {@link DoFn}
- * @param <FnOutputT> the output type of the {@link DoFn}
+ * @param <OutputT> the output type of the {@link DoFn}
  * @param <OutputT> the output type of the operator, this can be different from the fn output
```
Comment (aljoscha) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java]: Leftover Javadoc, I'll remove that while merging.

**Comment by aljoscha** [comment]
Thanks for the quick response, @JingsongLi. I'll merge as soon as Jenkins gives the green again. 😃

**Comment by aljoscha** [comment]
Run Flink ValidatesRunner

**Comment by aljoscha** [comment]
I merged, thanks a lot for working on this! 👍

**Pull request closed without merging**